### ISSUE TITLE
Extension of parallel/distributed backend scope

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#5380]
Following the initial refactor of parallel backends to a central location, to enable use of multiple backends, we can in-principle unlock multi-backend functionality in multiple locations.

The most direct entry points have been listed by @hazrulakmal in #5268, below.

Good first issue, a new contributor to this issue should pick one location to investigate (post here) and possibly refactor. Kindly note, it is not 100% clear whether all locations are easy to deal with, but it would be my suspicion that yes.

---
@hazrulakmal's list, from #5268:
---

I checked the codebase by searching `Parallel` and found many places where parallisation on joblib is done so perhaps when this PR is done and merged we can incrementally refactor them as well. Some of them to keep in mind, but not limited to, include the following:

1. https://github.com/sktime/sktime/blob/6709be6a1a804b708807dad7e4e191369edb1c66/sktime/classification/dictionary_based/_boss.py#L13
2. https://github.com/sktime/sktime/blob/6709be6a1a804b708807dad7e4e191369edb1c66/sktime/forecasting/ets.py#L14
3. https://github.com/sktime/sktime/blob/6709be6a1a804b708807dad7e4e191369edb1c66/sktime/transformations/panel/shapelet_transform.py#L1190
4. https://github.com/sktime/sktime/blob/6709be6a1a804b708807dad7e4e191369edb1c66/sktime/regression/interval_based/_tsf.py#L137
5. https://github.com/sktime/sktime/blob/6709be6a1a804b708807dad7e4e191369edb1c66/sktime/series_as_features/base/estimators/_ensemble.py#L234
6. https://github.com/sktime/sktime/blob/6709be6a1a804b708807dad7e4e191369edb1c66/sktime/classification/distance_based/_proximity_forest.py#L16
7. https://github.com/sktime/sktime/blob/6709be6a1a804b708807dad7e4e191369edb1c66/sktime/transformations/panel/supervised_intervals.py#L202

This is not an exhaustive list but one could check where else parallisation is done by searching `Parallel` on github search bar.

_Originally posted by @hazrulakmal in https://github.com/sktime/sktime/issues/5268#issuecomment-1734774495_

**Comment by tanish1729** [comment]
Hello! This seems to be a very old issue but I would be willing to take it up if still required.
I would like to start off by looking at the `shapelet_transform.py`   and updating the joblib Parallelisation with the sktime util (but as I understand, the sktime util is also written with the joblib Parallel being called).
Could you just clarify whether step 3 (exposing parameters to constructors) is what is being done in this file (``ForecastingGridSearchCV``)here 
![image](https://github.com/sktime/sktime/assets/92108099/d9554127-ff22-41a5-8353-612f2eb31160)

**Comment by fkiraly** [comment]
> This seems to be a very old issue but I would be willing to take it up if still required.

Not *that* old - it's still relevant though, thanks!

> but as I understand, the sktime util is also written with the joblib `Parallel` being called

yes, that's one of multiple backend options.

> Could you just clarify whether step 3 (exposing parameters to constructors) is what is being done in this file (`ForecastingGridSearchCV`) here

Yes, that's `backend` and `backend_params`, while `n_jobs` is being removed from the `__init__`.

**Comment by tanish1729** [comment]
Oh great. So using the sktime util, do I have to use different backends or can i use the `joblib Parallel` . I could make the changes for all the occurences then if there's nothing specific to be done differently.

**Comment by fkiraly** [comment]
the `sktime` utility just allows to select any integrated backend, not just `joblib`. Currently there´s only `dask` and "no parallelization", but new backends are easily added.

**Comment by tanish1729** [comment]
Great! I think now I have a good idea of what to do. I'll ping you on discord and ask a few queries I have related to actually contributing?

**Comment by julian-fong** [comment]
hey Franz, I would like to try refactoring _proximity_forest

If we are passing a parallel backend function for proximity forest, is the correct intuition that `fun=self._distance_to_exemplars_inst`, `iter = X.shape[0]`, and `meta = {"X_exemplar":self.X_exemplar, "instance": X.iloc[x, :], "distance_measure" = self._distance_measure}`? 

My question is also, will the para_func/para_dict[backend_name] be able to correctly deduce the correct value of 'instance' of x in X.iloc[x, :] when passing the meta dict into `for x in iter`? (hope my question makes sense!)?

**Comment by julian-fong** [comment]
is ```classification.tests.test_all_classifiers.py``` the right script to ensure all the classifier with refactored code works correctly?

**Comment by fkiraly** [comment]
Yes, that's part of it. There is a convenience method that collects all general interface tests for an object, it is `check_estimator` from `sktime.utils`, see more details here: https://www.sktime.net/en/latest/developer_guide/add_estimators.html#using-the-check-estimator-utility
